inbox/models/folder.py

```diff
@@ -76,10 +76,17 @@ def find_or_create(cls, session, account, name, role=None):
                      .format(name, account.id))
             raise
 
-        if not obj.canonical_name:
+        return obj
+
+
+    @classmethod
+    def update_role(cls, obj, role=''):
```

Comment:
As long as we have both folder and category in the codebase, I think the best way to keep the two in sync might be to override the setters. That way, people don't have to know `update_role` exists, and it's not possible for them to accidentally assign the role in a way that makes things fall out of sync.

So for example, we could say:

```
    _role = Column(String(MAX_INDEXABLE_LENGTH), nullable=False,
                            default='', name="canonical_name")

    @property
    def role(self):
        return self._role

    @role.setter
    def role(self, value):
        self._role = value
        self.category.name = value
```